ext/pg_result.c

```diff
@@ -1436,10 +1464,70 @@ pgresult_stream_each_tuple(VALUE self)
 	return pgresult_stream_any(self, yield_tuple);
 }
 
+/*
+ * call-seq:
+ *    res.field_name_type = Symbol
+ *
+ * Set type of field names specific to this result.
+ * It can be set to one of:
+ * * +:string+ to use String based field names
+ * * +:symbol+ to use Symbol based field names
+ * * +:static_symbol+ to use pinned Symbol (can not be garbage collected) - Don't use this, it will probably removed in future.
```

Comment:
Why deprecate `:static_symbol` the moment it is introduced? Is there a use case for having it?

Reply:
`:static_symbol` is the classic way to create symbols from C and it is also used by mysql2 gem. So I wanted to be able compare and benchmark it a bit more. So far I think it is mostly obsolete.